src/workerd/io/tracer.c++

```diff
@@ -117,6 +125,9 @@ void WorkerTracer::addLog(kj::Date timestamp, LogLevel logLevel, kj::String mess
     return;
   }
   trace->bytesUsed = newSize;
+  KJ_IF_SOME(writer, maybeTailStreamWriter) {
+    writer->report(context, tracing::Mark(tracing::Log(timestamp, logLevel, kj::str(message))));
+  }
```

Comment:
Perhaps consider adding `TODO` comments in here to make it clear that additional refactors are expected to handle the case where a worker has a streaming tail worker but no legacy tail worker.